official/projects/fffner/fffner_encoder.py

```diff
@@ -0,0 +1,319 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transformer-based BERT encoder network for FFFNER."""
+# pylint: disable=g-classes-have-attributes
+
+from typing import Any, Callable, Optional, Union
+from absl import logging
+import tensorflow as tf
+
+from official.nlp.modeling import layers
+
+
+_Initializer = Union[str, tf.keras.initializers.Initializer]
+_Activation = Union[str, Callable[..., Any]]
+
+_approx_gelu = lambda x: tf.keras.activations.gelu(x, approximate=True)
+
+@tf.keras.utils.register_keras_serializable(package='Text')
+class FFFNerEncoder(tf.keras.Model):
```

Comment:
Can we use keras layer instead and use the subclass style?
For example: https://github.com/tensorflow/models/blob/master/official/nlp/modeling/networks/bert_encoder.py#L31